cmd/object-handlers_test.go

```diff
@@ -438,3 +440,171 @@ func testAPICopyObjectHandler(obj ObjectLayer, instanceType, bucketName string,
 		}
 	}
 }
+
+// Wrapper for calling NewMultipartUpload tests for both XL multiple disks and single node setup.
+// First register the HTTP handler for NewMutlipartUpload, then a HTTP request for NewMultipart upload is made.
+// The UploadID from the response body is parsed and compared with the uploadID present in the upload.json for the same object.
+func TestAPINewMultipartHandler(t *testing.T) {
+	ExecObjectLayerAPITest(t, testAPINewMultipartHandler, []string{"CopyObject"})
+}
+
+func testAPINewMultipartHandler(obj ObjectLayer, instanceType, bucketName string, apiRouter http.Handler,
+	credentials credential, t TestErrHandler) {
+
+	objectName := "test-object-new-multipart"
+	rec := httptest.NewRecorder()
+	// construct HTTP request for copy object.
+	req, err := newTestSignedRequest("POST", getNewMultipartURL("", bucketName, objectName), 0, nil, credentials.AccessKeyID, credentials.SecretAccessKey)
+
+	if err != nil {
+		t.Fatalf("Failed to create HTTP request for copy Object: <ERROR> %v", err)
+	}
+	// Since `apiRouter` satisfies `http.Handler` it has a ServeHTTP to execute the logic of the handler.
+	// Call the ServeHTTP to executes the registered handler.
+	apiRouter.ServeHTTP(rec, req)
+	// Assert the response code with the expected status.
+	if rec.Code != http.StatusOK {
+		t.Fatalf("%s:  Expected the response status to be `%d`, but instead found `%d`", instanceType, http.StatusOK, rec.Code)
+	}
+	// decode the response body.
+	decoder := xml.NewDecoder(rec.Body)
+	multipartResponse := &InitiateMultipartUploadResponse{}
+
+	err = decoder.Decode(multipartResponse)
+	if err != nil {
+		t.Fatalf("Error decoding the recorded response Body")
+	}
+	// verify the existance and correctness of upload.json.
+	var uploadsJSON uploadsV1
+	switch v := obj.(type) {
+	case xlObjects:
+		// read the upload.json for xl backend.
+		uploadsJSON, err = v.readUploadsJSON(bucketName, objectName)
```

Comment:
We should only validate indeed if the response was correct we don't need to see if the backend was read properly.  In this situation we are skipping locking and all other forms of sanity checks leading to a bad test.